src/sw/redis++/redlock.hpp

```diff
@@ -0,0 +1,103 @@
+/**************************************************************************
+   Copyright (c) 2019
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ *************************************************************************/
+
+#ifndef REDISPLUSPLUS_REDLOCK_H
+#define REDISPLUSPLUS_REDLOCK_H
+
+#include "redis++.h"
+#include <unistd.h>
+#include <openssl/rc4.h>
+
+namespace sw {
+
+namespace redis {
+
+class RandomBufferInterface
+{
+public:
+	virtual ~RandomBufferInterface() {}
+	virtual std::string getUpdatedString() = 0;
+};
+
+template <typename RedisInstance>
```

Comment:
It doesn't need to be a template. In fact, the original RedLock algorithm only try to solve the distributed lock problem with a single Redis instance or multiple independent Redis instances.  If you do need to lock with Redis master. You can ask the user to get the Redis object with `RedisCluster::redis(hash_tag)` method, and pass it to `Redlock`.

Reply:
Thanks, you're right. I now have "RedLock" as a standard class in PR #33.